signer/src/eth.rs

```diff
@@ -0,0 +1,346 @@
+        }
+        /// Equivalent to `{DEV_PHRASE}//Ferdie`.
+        pub fn ferdie() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Ferdie").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//One`.
+        pub fn one() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//One").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Two`.
+        pub fn two() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Two").unwrap()).unwrap()
+        }
+    }
+}
+
+#[cfg(feature = "subxt")]
+mod subxt_compat {
+    use super::*;
+
+    impl<T: subxt::Config> subxt::tx::Signer<T> for Keypair
+    where
+        T::AccountId: From<AccountId20>,
+        T::Address: From<AccountId20>,
+        T::Signature: From<Signature>,
+    {
+        fn account_id(&self) -> T::AccountId {
+            self.account_id().into()
+        }
+
+        fn address(&self) -> T::Address {
+            self.account_id().into()
+        }
+
+        fn sign(&self, signer_payload: &[u8]) -> T::Signature {
+            self.sign(signer_payload).into()
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use proptest::prelude::*;
+
+    use hex::ToHex;
+    enum StubEthRuntimeConfig {}
+
+    impl subxt::Config for StubEthRuntimeConfig {
+        type Hash = subxt::utils::H256;
+        type AccountId = super::AccountId20;
+        type Address = super::AccountId20;
+        type Signature = super::Signature;
+        type Hasher = subxt::config::substrate::BlakeTwo256;
+        type Header =
+            subxt::config::substrate::SubstrateHeader<u32, subxt::config::substrate::BlakeTwo256>;
+        type ExtrinsicParams = subxt::config::SubstrateExtrinsicParams<Self>;
+        type AssetId = u32;
+    }
+
+    type Signer = dyn subxt::tx::Signer<StubEthRuntimeConfig>;
+
+    prop_compose! {
+        fn keypair()(seed in any::<[u8; 32]>()) -> Keypair {
+            let secret = SecretKey::from_slice(&seed).expect("valid secret key");
+            let inner = secp256k1::Keypair::from_secret_key(
+                &Secp256k1::new(),
+                &secret,
+            );
+
+            Keypair(ecdsa::Keypair(inner))
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn check_subxt_signer_implementation_matches(keypair in keypair(), msg in ".*") {
+            let msg_as_bytes = msg.as_bytes();
+
+            assert_eq!(Signer::account_id(&keypair), keypair.account_id());
+            assert_eq!(Signer::sign(&keypair, msg_as_bytes), keypair.sign(msg_as_bytes));
+        }
+
+        #[test]
+        fn check_account_id(keypair in keypair()) {
+            let account_id = {
+                let uncompressed = keypair.0.0.public_key().serialize_uncompressed();
+                let hash = keccak(&uncompressed[1..]).0;
+                let hash20 = hash[12..].try_into().expect("should be 20 bytes");
+                AccountId20(hash20)
```

Comment:
By having this code here, the assumption is that it is a known good way to create account IDs. Perhaps link to the moonbeam source or something where they do the same to show why this is the correct way?

Reply:
https://github.com/paritytech/subxt/issues/1506